app/services/SubmissionService.scala

```diff
@@ -73,51 +76,60 @@ class SubmissionService @Inject()(
       case Departure => connector.sendDepartureDeclaration(data.toXml)
     }
 
-  private def auditData(data: InventoryLinkingMovementRequest, result: String)(
-    implicit request: JourneyRequest[_]
-  ): Map[String, String] =
-    Map(
-      EORI.toString -> request.authenticatedRequest.user.eori,
-      MessageCode.toString -> data.messageCode,
-      UCRType.toString -> data.ucrBlock.ucrType,
-      UCR.toString -> data.ucrBlock.ucr,
-      MovementReference.toString -> data.movementReference.getOrElse(""),
-      SubmissionResult.toString -> result
-    )
-
   def submitDucrAssociation(
     mucrOptions: MucrOptions,
     associateDucr: AssociateDucr
-  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Int] = {
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext, request: JourneyRequest[_]): Future[Int] = {
     val timer = metrics.startTimer(Choice.AllowedChoiceValues.AssociateDUCR)
     connector
       .sendAssociationRequest(buildAssociationRequest(mucr = mucrOptions.mucr, ducr = associateDucr.ducr).toString)
       .map(_.status)
       .andThen {
-        case Success(_) =>
+        case Success(status) =>
+          auditService.audit(
+            Choice(Choice.AllowedChoiceValues.AssociateDUCR),
+            auditService.auditAssociateData(
+              request.authenticatedRequest.user.eori,
+              mucrOptions.mucr,
+              associateDucr.ducr,
+              status.toString
+            )
+          )
           timer.stop()
           metrics.incrementCounter(Choice.AllowedChoiceValues.AssociateDUCR)
       }
   }
 
   def submitDucrDisassociation(
     disassociateDucr: DisassociateDucr
-  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Int] = {
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext, request: AuthenticatedRequest[_]): Future[Int] = {
     val timer = metrics.startTimer(Choice.AllowedChoiceValues.DisassociateDUCR)
     connector
       .sendDisassociationRequest(buildDisassociationRequest(disassociateDucr.ducr).toString)
       .map(_.status)
       .andThen {
-        case Success(_) =>
+        case Success(status) =>
+          auditService.audit(
+            Choice(Choice.AllowedChoiceValues.DisassociateDUCR),
+            auditService
+              .auditDisassociateData(request.user.eori, disassociateDucr.ducr, status.toString)
```

Comment:
Would it not make sense to have `auditService.auditDisassociateData(...)` actually call `auditService.audit` under the hood?

Reply:
Very good point! @PatrykRudnicki suggested just that as well

Reply:
This has been addressed